src/main/java/mate/academy/bookstore/repository/BookSpecificationProviderManager.java

```diff
@@ -0,0 +1,22 @@
+package mate.academy.bookstore.repository;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.model.Book;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        System.out.println(bookSpecificationProviders);
+        return bookSpecificationProviders.stream()
+                .filter(p -> p.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Can't find correct "
```

Comment:
better to throw more specific exception where possible, let's go for NoSuchElement here for example